config_option_database/utils/test/test_Yacc2Graph.py

```diff
@@ -0,0 +1,62 @@
+#############################################################################
+# Copyright (c) 2019 Balabit
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+
+import pytest
+
+from utils.Yacc2Graph import _yacc2xml
+
+test_string = r"""
+%token test1
+%token test1next
+%token test2
+%token test2next
+%token KW_TEST
+%token number
+%token string
+%%
+start
+    : test
+    ;
+test
+    : test1 test1next
+    | test2 test2next test
+    | KW_TEST '(' test_opts ')'
+      {
+        int dummy_variable;
+        some_c_code(dummy_variable);
+      }
+    |
+    ;
+test_opts
+    : number
+    | string
+    ;
+%%
+"""
+
+def test_yacc2xml():
+    assert _yacc2xml(test_string)
+
+def test_failed_yacc2xml():
+    with pytest.raises(Exception) as e:
+        _yacc2xml(test_string[2:])
```

Comment:
We drop 2 characters to make an invalid yacc syntax, first is newline, second makes it invalid.

Reply:
Please `wrap test_string[2:]`.
For example: `generate_invalid_input(test_string)`.

Or even better, this would be enough?
```
 _yacc2xml("somethingthatisinvalid")
```
?

Reply:
Good idea, I will use your second tip.

Reply:
Done!